### PR TITLE
0.14.0 migration guide

### LINKED ISSUE #687: Migration guide for new data transformer
### Feature Request

.

### PR DESCRIPTION
**Pull request by cptartur**
Closes #687

## Introduced changes

- This PR adds initial version of 0.14.0 migration guide.
- This PR changes ContractDeployment NamedTuple to use class syntax for documentation purposes. It should be functionally equivalent to the old code and shouldn't be breaking at all.

##

- [ ] This PR contains breaking changes

### CODE REVIEW

File: docs/migration_guide.rst [hunk 1]
```diff
@@ -1,6 +1,30 @@
...
+Breaking changes
+----------------
+
+1. Renamed :meth:`~starknet_py.net.udc_deployer.deployer.ContractDeployment` first parameter from ``udc`` to ``call``.
+
+Deprecations
+------------
+
+1. ``compiler`` module. It will be removed in the future. We recommend transitioning
+    to building contracts through StarkNet CLI or external tools and using only compiled contracts with StarkNet.py
```
Comment (war-in) [inline docs/migration_guide.rst:19]:
```suggestion
    to building contracts through StarkNet CLI or external tools and using only compiled contracts with StarkNet.py.
```

File: docs/migration_guide.rst [hunk 2]
```diff
@@ -1,6 +1,30 @@
...
+**********************
+0.14.0 Migration guide
+**********************
+
+This version deprecates several modules and fixes underlying issues with several others.
+
+Breaking changes
+----------------
+
+1. Renamed :meth:`~starknet_py.net.udc_deployer.deployer.ContractDeployment` first parameter from ``udc`` to ``call``.
```
Comment (war-in) [inline docs/migration_guide.rst:13]:
```suggestion
1. Renamed :meth:`~starknet_py.net.udc_deployer.deployer.ContractDeployment` (one returned by ``Deployer.create_deployent_call``) first parameter from ``udc`` to ``call``.
```

**Review by war-in — APPROVED** [review]

File: docs/migration_guide.rst [hunk 3]
```diff
@@ -1,6 +1,30 @@
...
+
+Deprecations
+------------
+
+1. ``compiler`` module. It will be removed in the future. We recommend transitioning
+    to building contracts through StarkNet CLI or external tools and using only compiled contracts with StarkNet.py
+2. ``utils.data_transformer`` module. It has been replaced with ``serializer`` module.
```
[the remark this answers is not in the document]
Comment (cptartur) [reply docs/migration_guide.rst:20]: Serializer is linked below in more details, but okay.

File: docs/migration_guide.rst [hunk 4]
```diff
@@ -1,6 +1,30 @@
 Migration guide
 ===============
 
+**********************
+0.14.0 Migration guide
+**********************
```
[the remark this answers is not in the document]
Comment (cptartur) [reply docs/migration_guide.rst:6]: What do you mean exactly?

[the remark this answers is not in the document]
Comment (drknzz) [reply docs/migration_guide.rst:6]: As do we want to have the latest version migration guide on the top level of the side bar

[the remark this answers is not in the document]
Comment (cptartur) [reply docs/migration_guide.rst:6]: I think having single `Migration guide` on sidebar is good enough.

**Review by drknzz — APPROVED** [review]

**Pull request merged**